### PR TITLE
modules: config: add keyboard module

### PR DESCRIPTION
**Pull request by rissson**

### CODE REVIEW

File: modules/config/keyboard.nix [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      };
+
+      keyMap = mkOption {
+        type = types.str;
+        example = "us";
+        description = ''
+          The keyboard mapping table for the virtual consoles. The default is
+          the layout associated with the keyMap but you may need to change this
+          depending on your keyboard variant.
+        '';
```
Comment (kalbasit) [inline modules/config/keyboard.nix:30]: Proposal: Let's do a section separation here between x11 settings and console settings. Also what section would fit best Wayland?

File: modules/config/keyboard.nix [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+with lib;
+let
+  cfg = config.soxin.settings.keyboard;
+
+  layoutModule = types.submodule ({ config, ... }: {
+    options = {
+      layout = mkOption {
+        type = types.str;
+        default = "us";
```
Comment (kalbasit) [inline modules/config/keyboard.nix:11]: If you pick the config of NixOS here as the default, then it should help people migrate over.

**Review by kalbasit — APPROVED** [review]

File: modules/config/keyboard.nix (hunk 2 above)
Comment (rissson) [reply modules/config/keyboard.nix:11]: The problem with using NixOS (or home-manager) configuration as a default here is that they both are comma-separated strings (for layout and variant). Iterating over both of those and settings our config (list of attributes) would be too much of a pain, and I don't think it's asking a lot from people who are migrating to setup their keyboard configuration.

File: modules/config/keyboard.nix (hunk 1 above)
Comment (rissson) [reply modules/config/keyboard.nix]: I'm not sure about wayland. I guess that's a question for later, as we add graphical stuff.

**Pull request merged**